packages/wdio-utils/src/shim.ts

```diff
@@ -160,27 +324,51 @@ async function executeAsync(this: any, fn: Function, retries: Retries, args: any
 let executeSync = executeSyncFn
 
 /**
- * shim to make sure that we only wrap commands if wdio-sync is installed as dependency
+ * Method to switch between sync and async execution. It allows to have async
+ * tests in between synchronous tests. `fn` can either return a promise (e.g. for `executeSync`)
+ * or a function (e.g. for `runSync`). In both cases we need to make sure that
+ * we flip `runAsync` flag to true to that commands are wrapped with the @wdio/sync
+ * wrapper.
  */
-try {
-    /**
-     * only require `@wdio/sync` if `WDIO_NO_SYNC_SUPPORT` which allows us to
-     * create a smoke test scenario to test actual absence of the package
-     * (internal use only)
-     */
-    /* istanbul ignore if */
-    if (!process.env.WDIO_NO_SYNC_SUPPORT) {
-        const packageName = '@wdio/sync'
-        const wdioSync = require(packageName)
-        hasWdioSyncSupport = true
-        runFnInFiberContext = wdioSync.runFnInFiberContext
-        wrapCommand = wdioSync.wrapCommand
-        executeHooksWithArgs = wdioSync.executeHooksWithArgs
-        executeSync = wdioSync.executeSync
-        runSync = wdioSync.runSync
+export function switchSyncFlag (fn: Function) {
+    return function (this: unknown, ...args: any[]) {
+        const switchFlag = runAsync
+        runAsync = false
+        const result = fn.apply(this, args)
+
+        if (typeof result.finally === 'function') {
+            return result.finally(() => (runAsync = switchFlag))
+        }
+
+        if (typeof result === 'function') {
+            return function (this: any, ...args: any[]) {
+                const switchFlag = runAsync
+                const res = result.apply(this, args)
+                if (typeof result.finally === 'function') {
+                    return result.finally(() => (runAsync = switchFlag))
```

Comment:
no need to return here right?
```suggestion
                    return result.finally(() =>  { runAsync = switchFlag })
```